### PR TITLE
adding new way to configure test services

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: test/WebJobs.Script.Tests.Integration/TestScriptHostBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    // When skipping host startup (e.g. offline), we need most general services registered so admin
+                    // APIs can function. However, we want to prevent the ScriptHost from
+                    // actually starting up. To accomplish this, we remove the host service
+                    // responsible for starting the job host.
+                    var jobHostService = services.FirstOrDefault(p => p.ImplementationType == typeof(JobHostService));
+                    services.Remove(jobHostService);
+                });
+            }
+
+            _configureTestServices(builder);
```
Comment (brettsam) [inline test/WebJobs.Script.Tests.Integration/TestScriptHostBuilder.cs]: This is what we need.

File: test/WebJobs.Script.Tests.Integration/TestScriptHostBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+using System.Linq;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Integration
+{
+    internal class TestScriptHostBuilder : IScriptHostBuilder
```
Comment (brettsam) [inline test/WebJobs.Script.Tests.Integration/TestScriptHostBuilder.cs]: This is a copy/paste of DefaultScriptHostBuilder with the configurable Action called before Build(). That class is sealed (by design?), or I'd add a virtual method and derive from it. But I'm not sure we want to make it easy to do that...

**Comment by fabiocav** [comment]
@brettsam based on the conversations we had, should this be closed?

**Comment by fabiocav** [comment]
Closing this. Please rebase and reopen if this is still relevant.

**Pull request closed without merging**